internal/mode/static/state/graph/backend_refs.go

```diff
@@ -288,16 +307,39 @@ func getServiceAndPortFromRef(
 
 	svc, ok := services[svcNsName]
 	if !ok {
-		return svcNsName, v1.ServicePort{}, field.NotFound(refPath.Child("name"), ref.Name)
+		return svcNsName, v1.ServicePort{}, []v1.IPFamily{}, field.NotFound(refPath.Child("name"), ref.Name)
 	}
 
 	// safe to dereference port here because we already validated that the port is not nil in validateBackendRef.
 	svcPort, err := getServicePort(svc, int32(*ref.Port))
 	if err != nil {
-		return svcNsName, v1.ServicePort{}, err
+		return svcNsName, v1.ServicePort{}, []v1.IPFamily{}, err
 	}
 
-	return svcNsName, svcPort, nil
+	svcIPFamilies := svc.Spec.IPFamilies
+
+	return svcNsName, svcPort, svcIPFamilies, nil
+}
+
+func verifyIPFamily(npCfg *NginxProxy, svcIPFamily []v1.IPFamily) error {
+	if npCfg.Source == nil {
+		return nil
+	}
+
+	// we can access this field since we have already validated that ipFamily is not nil in validateNginxProxy.
```

Comment:
in validateNginxProxy, we don't do anything if it is nil -- https://github.com/nginxinc/nginx-gateway-fabric/blob/1481d4df9a00571c4292545a1cf27cdafab70013/internal/mode/static/state/graph/nginxproxy.go#L111

so in that case, let's assume Dual family